src/main/java/org/pesmypetcare/webservice/dao/communitymanager/ForumDaoImpl.java

```diff
@@ -137,36 +147,69 @@ public void updateTags(String parentGroup, String forumName, List<String> newTag
     }
 
     @Override
-    public void postMessage(String parentGroup, String forumName, MessageEntity messageEntity)
+    public void postMessage(String parentGroup, String forumName, Message message)
         throws DatabaseAccessException, DocumentException {
         String groupId = groupDao.getGroupId(parentGroup);
         String forumId = getForumId(parentGroup, forumName);
-        messageEntity.setPublicationDate(timeFormatter.format(LocalDateTime.now()));
-        documentAdapter.createDocument(Path.ofCollection(Collections.messages, groupId, forumId), messageEntity);
+        MessageEntity messageEntity = new MessageEntity(message);
+        WriteBatch batch = documentAdapter.batch();
+        documentAdapter.createDocument(Path.ofCollection(Collections.messages, groupId, forumId), messageEntity, batch);
+        /*ImageEntity imageEntity = message.getImage();
```

Comment:
It is necessary to have commented code?

Reply:
It was there in case we decided to go back to the previous implementation of the image post. But you're right, it should'nt be here. In case we need it in the futurue we can just get it from previous commits.